likecoin/admin/ajax.php

```diff
@@ -75,10 +75,17 @@ function likecoin_matters_login() {
 	if ( ! isset( $_POST[ LC_OPTION_MATTERS_ID_FIELD ] ) || ! isset( $_POST[ LC_OPTION_MATTERS_PASSWORD_FIELD ] ) ) {
 		wp_send_json_error( array( 'error' => 'MISSING_FIELDS' ) );
 	}
-	$matters_id       = $_POST[ LC_OPTION_MATTERS_ID_FIELD ];
-	$matters_password = $_POST[ LC_OPTION_MATTERS_PASSWORD_FIELD ];
-	$results          = LikeCoin_Matters_API::get_instance()->login( $matters_id, $matters_password );
-	wp_send_json( $results );
+	$matters_id           = $_POST[ LC_OPTION_MATTERS_ID_FIELD ];
+	$matters_password     = $_POST[ LC_OPTION_MATTERS_PASSWORD_FIELD ];
+	$results              = LikeCoin_Matters_API::get_instance()->login( $matters_id, $matters_password );
+	$matters_access_token = isset( $results['data']['userLogin']['token'] ) ? $results['data']['userLogin']['token'] : null;
+	$user_info_results    = array();
+	if ( isset( $matters_access_token ) ) {
+		$user_info_results = LikeCoin_Matters_API::get_instance()->query_user_info( $matters_access_token );
+		wp_send_json( array_merge( $results['data'], array( 'viewer' => $user_info_results ) ) );
```

Comment:
Not familiar with PHP, but seems the two `wp_send_json` are sending data with different schema?
One is `$results['data']` while another is `$results`?

Reply:
yes, its a hack to prevent the 2 arrays both wrapped in `data` key to be overwritten when merging
good response looks like `{ data: { action: paylod }}` and errors are like `{errors: [], data: null }`